components/script/dom/audiotracklist.rs

```diff
@@ -102,17 +97,16 @@ impl AudioTrackList {
         );
     }
 
+
     pub fn add(&self, track: &AudioTrack) {
         self.tracks.borrow_mut().push(Dom::from_ref(track));
         track.add_track_list(self);
     }
 
     pub fn clear(&self) {
-        self.tracks
-            .borrow()
-            .iter()
-            .for_each(|t| t.remove_track_list());
-        self.tracks.borrow_mut().clear();
+        let mut tracks = self.tracks.borrow_mut();
+        tracks.iter().for_each(|t| t.remove_track_list());
+        tracks.clear();
```

Comment:
I think the previous version of this was better...the mutable borrow was scoped to the line that used it.